### ISSUE TITLE
[FEATURE] - Add option to mass export download url's for use in third party downloaders

### ISSUE THREAD

**Issue by aniel300** [PlexRipper/PlexRipper#105]
so taking a look at "https://github.com/PlexRipper" i can see in the description after the word "everything" it has a "*" but i dont think think it is explained as to what that means, however the reason am making this post is to find out if the is a option to download everything not because am trying to abuse or anything but because i want to create a starting point from a friend plex server and then continue with radarr and sonarr, thank you in advance.

**Comment by RXWatcher** [comment]
You could just ask your 'friend' for a copy of his content by other means like sftp? I know if I saw a 'friend' downloading all of the content that I spent years collecting that I'd ban them in a second..even if they were by best 'friend'. They could call me on the phone and we could arrange something.

**Comment by aniel300** [comment]
in my case i can't ask that friend for sftp access

**Comment by JasonLandbridge** [comment]
You're free to download everything you want with PlexRipper, at your own risk of course. Will close this due to age

**Comment by aniel300** [comment]
@JasonLandbridge can u do mass downloads with the latest versions?

**Comment by JasonLandbridge** [comment]
Yes! Although it is a bit buggy, especially downloading lots of tv shows, its definitely a possibility. There is also the issue that large libraries will freeze the ui due to all the data that being loaded in at once but it's being worked on. 

In the short term, I would also like to implement something that will generate download urls and then you can import those urls into something like JDownloader2 and download that way. Also see https://github.com/PlexRipper/PlexRipper/issues/93 for some idea's related to this

**Comment by aniel300** [comment]
yes i can live with manually adding the urls to jdownloader, will check the worksarounds later, thank u.

**Comment by JasonLandbridge** [comment]
My pleasure, quick question: Do you know what url format JDownloader would like that also includes the filename?

Something like: [Name] - [DownloadUrl] ?

I'm asking because the files that are downloaded from a Plex server have a cryptic name so ideally JDownloader would automatically name the file that comes from the download url

**Comment by aniel300** [comment]
hold on let me do a quick manual check

**Comment by aniel300** [comment]
ok so i did a quick test comparing jd and idm and idm can get name but jd can not get download file/name even when providing direct url generated by idm. not sure if this is what u asked, if u can explain more i can try to help. 
![IDMan_LFZmPMX8ls](https://user-images.githubusercontent.com/7571729/208166559-687e002a-0911-4c46-887e-5b383609e39e.png)

**Comment by JasonLandbridge** [comment]
Yes that's what I'm looking for!

So let's say the url is:

`https://random-server/path/to/movie/file.mp4`

And I want to make sure it is saved as 

`my-little-pony.mp4`

How would I need to format all those urls such that the file is downloaded as `my-little-pony.mp4`?

**Comment by aniel300** [comment]
use idm so u cant get the temp url and so u can anylize it, if u dont know let me know and i can share one

**Comment by Hemps38** [comment]
> ok so i did a quick test comparing jd and idm and idm can get file/name but not year. jd can not get download file/name even when providing direct url generated by idm. not sure if this is what u asked, if u can explain more i can try to help. ![IDMan_LFZmPMX8ls](https://user-images.githubusercontent.com/7571729/208166852-adcf4d31-9257-492c-b89c-2341c1d18c78.png)

I manage to get the link but IDM still shows file.mkv

Any ideas?

**Comment by JasonLandbridge** [comment]
@Hemps38 Yeah that's how Plex stores every file on a server, it's always `file.mkv` or `file.mp4` etc

@aniel300 Ah now I understand that you mean "Internet Download Manager", but I'm not sure what the rest of your comment says.

> My pleasure, quick question: Do you know what url format JDownloader would like that also includes the filename?
> 
> Something like: [Name] - [DownloadUrl] ?
> 
> I'm asking because the files that are downloaded from a Plex server have a cryptic name so ideally JDownloader would automatically name the file that comes from the download url

So this question still stands, I need a text format that either JDownloader or IDM can use to import many links at once and then use the name next to it to save it as. As the download links are always `file.mkv` or `file.mp4` etc. I can't change the download link because it comes directly from Plex

**Comment by aniel300** [comment]
i dont think i know what u asking, i will do some more ts later

**Comment by aniel300** [comment]
> Yes! Although it is a bit buggy, especially downloading lots of tv shows, its definitely a possibility. There is also the issue that large libraries will freeze the ui due to all the data that being loaded in at once but it's being worked on.
> 
> In the short term, I would also like to implement something that will generate download urls and then you can import those urls into something like JDownloader2 and download that way. Also see [#93](https://github.com/PlexRipper/PlexRipper/issues/93) for some idea's related to this

as per this, am assuming and hoping this feature have been fully implemented and is working fine. i never got to use this project to the full potential but i will be looking into https://github.com/PlexRipper/PlexRipper/issues/264 to try to do so. thank devs for the hard work, closing for now.

**Comment by aniel300** [comment]
was this option ever added? am playing with this now but i dont see a way
